core/domain/classifier_services.py

```diff
@@ -347,6 +348,75 @@ def mark_training_job_complete(job_id):
                                            feconf.TRAINING_JOB_STATUS_COMPLETE)
 
 
+def mark_training_job_failed(job_id):
+    """Updates the training job's status to failed.
+
+    Args:
+        job_id: str. ID of the ClassifierTrainingJob.
+    """
+    _update_classifier_training_job_status(job_id,
+                                           feconf.TRAINING_JOB_STATUS_FAILED)
+
+
+def mark_training_job_pending(job_id):
+    """Updates the training job's status to pending.
+
+    Args:
+        job_id: str. ID of the ClassifierTrainingJob.
+    """
+    _update_classifier_training_job_status(job_id,
+                                           feconf.TRAINING_JOB_STATUS_PENDING)
+
+
+def update_failed_jobs(training_jobs):
+    """Updates status of job_models to failed.
+
+    Args:
+        training_jobs: list. List of jobs to be updated as failed.
+    """
+    for training_job in training_jobs:
+        mark_training_job_failed(training_job.job_id)
+
+
+def fetch_next_job():
+    """Gets next job model in the job queue.
+
+    Returns:
+        ClassifierTrainingJobModel. Domain object of the next training Job.
+    """
+    classifier_training_jobs = []
+    cursor = None
+    valid_jobs = []
+    failed_jobs = []
+
+    while len(valid_jobs) == 0:
```

Comment:
(1) Email sounds good to me, thanks! The only thing I'll say is that I think the emails should probably go to you and Anmol, since you'll know what to do more than me. (I don't mind being cc'd for info but I think you both would be better points of contact.) Let's make sure to file a separate issue for this so we don't forget to do it and end up running into surprises later.

(2) Yep! Though I realize I may have misnamed it -- the field sounds more like next_scheduled_check_time (or similar) to me.